Mathlib/Data/Finsupp/SMulWithZero.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Kim Morrison
+-/
+import Mathlib.Algebra.Group.Action.Pi
+import Mathlib.Algebra.SMulWithZero
+import Mathlib.Data.Finsupp.Defs
+
+/-!
+# Scalar multiplication on `Finsupp`
+
+This file defines the pointwise scalar multiplication on `Finsupp`, assuming it preserves zero.
```

Comment:
I'm assuming the choice not to define the `Module` instance yet here is a deliberate one; can you add an `assert_not_exists Module` and a sentence to the docstring explaining that this is why this file exists as it does?

Reply:
(I guess the reasoning is that the dependency chain is `SMulZeroClass <- AddCommGroup <- Module`?)